packages/desktop/views/dashboard/send-flow/views/InputTokenAmountView.svelte

```diff
@@ -3,64 +3,71 @@
     import {
         IAsset,
         NewTransactionType,
-        getAssetById,
+        formatTokenAmountDefault,
+        getUnitFromTokenMetadata,
         newTransactionDetails,
         updateNewTransactionDetails,
     } from '@core/wallet'
-    import { AssetAmountInput, Button, FontWeight, Text, TextType } from 'shared/components'
+    import { TokenAmountInput, TokenAmountTile } from '@ui'
     import { get } from 'svelte/store'
     import { sendFlowRouter } from '../send-flow.router'
+    import SendFlowTemplate from './SendFlowTemplate.svelte'
 
     const transactionDetails = get(newTransactionDetails)
-    let assetAmountInput: AssetAmountInput
+    let assetAmountInput: TokenAmountInput
     let asset: IAsset
     let rawAmount: string
     let amount: string
     let unit: string
-    let disableAssetSelection: boolean
 
     if (transactionDetails.type === NewTransactionType.TokenTransfer) {
-        asset = getAssetById(transactionDetails.assetId)
+        asset = transactionDetails.asset
         rawAmount = transactionDetails.rawAmount
-        unit = transactionDetails.unit
-        disableAssetSelection = transactionDetails.disableAssetSelection
+        unit = transactionDetails.unit || getUnitFromTokenMetadata(asset?.metadata)
     }
 
-    function onContinueClick(): void {
-        updateNewTransactionDetails({
-            type: NewTransactionType.TokenTransfer,
-            rawAmount,
-            unit,
-            disableAssetSelection,
-        })
-        $sendFlowRouter.next()
+    $: availableBalance = asset?.balance?.available
+
+    function setToMax(): void {
+        if (asset?.metadata?.decimals) {
+            amount = formatTokenAmountDefault(availableBalance, asset?.metadata, unit, false)
+        } else {
+            amount = availableBalance.toString() ?? '0'
+        }
+    }
+
+    async function validate(): Promise<boolean> {
+        try {
+            await assetAmountInput?.validate()
+            return true
+        } catch (err) {
+            return false
+        }
+    }
+
+    async function onContinueClick(): Promise<void> {
+        const isValid = await validate()
+        if (isValid) {
+            updateNewTransactionDetails({
+                type: NewTransactionType.TokenTransfer,
+                rawAmount,
+            })
+            $sendFlowRouter.next()
+        }
```

Comment:
You can get rid of the function above if you move the try catch here
```suggestion
    async function onContinueClick(): Promise<void> {
       try {
            await assetAmountInput?.validate()
            updateNewTransactionDetails({
                type: NewTransactionType.TokenTransfer,
                rawAmount,
            })
            $sendFlowRouter.next()
        } catch (err) {
           ...
        }
           
```